### ISSUE TITLE
Should we also track public hosted implementations?

### ISSUE THREAD

**Issue by blewis** [IQSS/open-source-at-harvard#3]
Perhaps we should have a column in github.tsv for projects which have public hosted implementations?  Perhaps another column for open APIs?  
Maybe another (if this isn't going too far) for implementations running on open source clouds like Massachusetts Open Cloud which runs Open Stack?

**Comment by pdurbin** [comment]
@blewis I love these ideas! Thanks! We should also have have a column for demo sites so people can kick the tires. [Right now](https://github.com/IQSS/open-source-at-harvard/blob/2f54f120724aac1604fa147424c330d180986169/github.tsv) github.tsv only has a single column called "Repo" but in the future it could look something like this:

| Repo  | Production URL hosted by Harvard | Demo URL |
| --- | --- | --- |
| https://github.com/cga-harvard/worldmap  | http://worldmap.harvard.edu  |
| https://github.com/IQSS/dataverse  | https://dataverse.harvard.edu | https://demo.dataverse.org |

@blewis what name would you give to the column for the Mass Open Cloud thing? That's the only one I'm fuzzy on.

**Comment by blewis** [comment]
Great!
Mass Open Cloud:  Thinking about it more, rather than getting into type of cloud host, something much more useful might be a link to the containerized version (which will often but not necessarily be Docker I think).  That column could be called "Container" perhaps.  
And maybe a column for the endpoint to any open API or documentation for it?

**Comment by pdurbin** [comment]
@blewis can you please send me a link to a containerized version of something?

Also, I forgot about the API stuff. At first I was thinking "Has API" with "true" vs "false" but more useful would probably be "API Guide" with a link like this:

| Repo  | Production URL hosted by Harvard | Demo URL | API Guide |
| --- | --- | --- | --- |
| https://github.com/cga-harvard/worldmap  | http://worldmap.harvard.edu  |
| https://github.com/IQSS/dataverse  | https://dataverse.harvard.edu | https://demo.dataverse.org | http://guides.dataverse.org/en/latest/api |

**Comment by blewis** [comment]
Here is the dockerhub endpoint for the docker images we at CGA maintain https://hub.docker.com/r/harvardcga/.  Some projects like Billion Object Platform(BOP) have multiple images so might be best to just point to the main harvardcga endpoint for all CGA projects. The three projects I mentioned, BOP, WorldMap, HHypermap Registry all have docker images.

**Comment by blewis** [comment]
Here is the self documenting API endpoint for the BOP http://bop.worldmap.harvard.edu/bopws/swagger#/default

**Comment by pdurbin** [comment]
@blewis how does this look? I'm using https://hub.docker.com/r/harvardcga/bop-ui/ because it has a reasonable description of the BOP rather than nothing. ( https://hub.docker.com/r/harvardcga/bop-webservice/ is the one I looked at first.) I hear you on multiple Docker images being necessary but I think the description is the most important thing.

| Repo  | Production URL | Demo URL | API Guide | Docker |
| --- | --- | --- | --- | --- |
| https://github.com/cga-harvard/hhypermap-bop  | https://hub.docker.com/r/ndslabs/dataverse/ | | http://bop.worldmap.harvard.edu/bopws/swagger#/default | https://hub.docker.com/r/harvardcga/bop-ui/ |
| https://github.com/IQSS/dataverse  | https://dataverse.harvard.edu | https://demo.dataverse.org | http://guides.dataverse.org/en/latest/api | https://hub.docker.com/r/ndslabs/dataverse/ |

**Comment by blewis** [comment]
looks good!

**Comment by pdurbin** [comment]
@blewis I have another crazy idea, up your alley since it's geospatial in nature. 😄 

What if we could make a map of where these projects are headquartered. What would be the best way to do this? Should we have a columns called "HQ Latitude" and "HQ Longitude"? Or is there a better way?

**Comment by pdurbin** [comment]
Other ideas of columns we could add that were mentioned as a "dev practices checklist" at https://irclogs.jackgrigg.com/irc.freenode.net/openhatch/2017-07-16#i_4157921 include:

- A link to a list of issues in code review, which is an indication that code review happens at all, such as https://github.com/IQSS/dataverse/labels/Status%3A%20Code%20Review
- A link to the latest automated test status on Travis CI or similar such as https://travis-ci.org/IQSS/dataverse
- A link to code-level documentation, such as http://phoenix.dataverse.org/site/latest/site/apidocs/
- A link to the latest code coverage, such as https://coveralls.io/github/IQSS/dataverse

**Comment by pdurbin** [comment]
I occurs to me that what we're talking about here is a need for a "consistent, agreed upon method of expressing project metadata in a vendor-neutral way than could be utilized by a variety of consumers" as mentioned at https://github.com/todogroup/project-metadata . I originally pinged @willnorris about this at https://github.com/IQSS/dataverse/issues/3736 and I guess I just did the same here. 😄  Feedback and ideas from all are very welcome!

**Comment by blewis** [comment]
Good idea. "HQ Latitude" and "HQ Longitude" would provide the data needed to easily throw projects onto a map.

**Comment by pdurbin** [comment]
@blewis in the case of Dataverse, we have not only an HQ installation but a number of installations around the world and the latitude and longitude for each of these installations. I'm wondering if you have any thoughts on how to work this data into this "open source at Harvard" dataset/project. I'd also love to show you a new take on our map of installations. I'm using different technology than we used in the past. Here's a screenshot:

<img width="610" alt="Screen Shot 2019-10-03 at 6 06 05 AM" src="https://user-images.githubusercontent.com/21006/66118395-48c86280-e5a4-11e9-9119-79ef2d3157dd.png">
